client/chain-spec/src/chain_spec.rs

```diff
@@ -160,6 +160,12 @@ struct ClientSpec<E> {
 	#[serde(skip_serializing)]
 	genesis: serde::de::IgnoredAny,
 	light_sync_state: Option<SerializableLightSyncState>,
+	/// Mapping from `block_hash` to `wasm_code`.
+	///
+	/// The given `wasm_code` will be used to substitute the on-chain wasm code from the given
+	/// block hash onwards.
+	#[serde(default)]
+	code_substitutes: HashMap<String, Bytes>,
```

Comment:
In a future pr I would like to replace `String` with something that directly checks that we got here some hex value.